app/src/org/commcare/views/widgets/GregorianDateWidget.java

```diff
@@ -370,7 +366,9 @@ private void refreshDisplay() {
 
     private void openCalendar() {
         setFocus(getContext());
-        timeBeforeCalendarOpened = calendar.getTimeInMillis();
+        timeBeforeCalendarOpened = getCurrentMillis();

```

Comment:
Fix number 2: The `timeBeforeCalendarOpened` should be set based on the current UI of the main widget, and the calendar that underlies the calendar fragment should be updated to use the same